shelley/chain-and-ledger/executable-spec/test/Generator/Core/QuickCheck.hs

```diff
@@ -0,0 +1,141 @@
+import           Crypto.Random (drgNewTest, withDRG)
+import           Data.Tuple (swap)
+import           Data.Word (Word64)
+
+import           Test.QuickCheck (Gen)
+import qualified Test.QuickCheck as QC
+
+import           Address (toAddr, toCred)
+import           Coin (Coin (..))
+import           Examples (mkKeyPair)
+import           Keys (pattern KeyPair, hashKey, vKey)
+import           LedgerState (pattern LedgerState, genesisCoins, genesisState)
+import           MockTypes (Addr, DPState, KeyPair, KeyPairs, LedgerEnv, SignKeyVRF, TxOut, UTxO,
+                     UTxOState, VKey, VerKeyVRF)
+import           Numeric.Natural (Natural)
+import           Tx (pattern TxOut)
+import           TxData (pattern AddrBase, pattern KeyHashObj)
+
+genBool :: Gen Bool
+genBool = QC.arbitraryBoundedRandom
+
+genInteger :: Integer -> Integer -> Gen Integer
+genInteger lower upper = QC.choose (lower, upper)
+
+-- | Generator for a natural number between 'lower' and 'upper'
+genNatural :: Natural -> Natural -> Gen Natural
+genNatural lower upper = fromInteger <$> QC.choose (lower', upper')
+ where
+  lower' = fromIntegral lower
+  upper' = fromIntegral upper
+
+mkKeyPairs :: Word64 -> (KeyPair, KeyPair)
+mkKeyPairs n
+  = (mkKeyPair_ (2*n), mkKeyPair_ (2*n+1))
+  where
+    mkKeyPair_ n_ = (uncurry KeyPair . swap) (mkKeyPair (n_,n_,n_,n_,n_))
+
+-- | Constant list of KeyPairs intended to be used in the generators.
+traceKeyPairs :: KeyPairs
+traceKeyPairs = mkKeyPairs <$> [1 .. 150]
+
+-- | Select between _lower_ and _upper_ keys from 'traceKeyPairs'
+someKeyPairs :: Int -> Int -> Gen KeyPairs
+someKeyPairs lower upper =
+  take
+    <$> QC.choose (lower, upper)
+    <*> QC.shuffle traceKeyPairs
+
+-- | Find first matching key pair for address. Returns the matching key pair
+-- where the first element of the pair matched the hash in 'addr'.
+findPayKeyPair :: Addr -> KeyPairs -> KeyPair
+findPayKeyPair (AddrBase (KeyHashObj addr) _) keyList =
+    case matches of
+      []    -> error "findPayKeyPair: could not find a match for the given address"
+      (x:_) -> fst x
+    where
+      matches = filter (\(pay, _) -> addr == hashKey (vKey pay)) keyList
+findPayKeyPair _ _ = error "findPayKeyPair: expects only AddrBase addresses"
+
+-- | Select one random verification staking key from list of pairs of KeyPair.
+pickStakeKey :: KeyPairs -> Gen VKey
+pickStakeKey keys = vKey . snd <$> QC.elements keys
+
+-- | Generates a list of coins for the given 'Addr' and produced a 'TxOut' for each 'Addr'
+genTxOut :: [Addr] -> Gen [TxOut]
+genTxOut addrs = do
+  ys <- genCoinList 100 10000 (length addrs) (length addrs)
+  return (uncurry TxOut <$> zip addrs ys)
+
+-- | Generates a list of 'Coin' values of length between 'lower' and 'upper'
+-- and with values between 'minCoin' and 'maxCoin'.
+genCoinList :: Integer -> Integer -> Int -> Int -> Gen [Coin]
+genCoinList minCoin maxCoin lower upper = do
+  len <- QC.choose (lower, upper)
```

Comment:
This is an example of numeric scaling behavior being different. Here we observe a constant range which doesn't scale with the Size parameter of QC.

Whereas here https://github.com/input-output-hk/cardano-ledger-specs/blob/c738c46d70f2c6ee69350d4b4c5aea6927c1dba2/shelley/chain-and-ledger/executable-spec/test/Generator/Core.hs#L94 we observe linear scaling with HH's Size parameter.